DigitalLearningSolutions.Web/Controllers/LoginController.cs

```diff
@@ -40,37 +41,84 @@ public IActionResult Index(LoginViewModel model)
             }
 
             var (verifiedAdminUser, verifiedDelegateUsers) =
-                loginService.VerifyUsers(model.Password, unverifiedAdminUser, unverifiedDelegateUsers);
+                loginService.VerifyUsers(model.Password, adminUser, delegateUsers);
 
             if (verifiedAdminUser == null && verifiedDelegateUsers.Count == 0)
             {
                 ModelState.AddModelError("Password", "The password you have entered is incorrect.");
                 return View("Index", model);
             }
 
-            var approvedDelegateUser = verifiedDelegateUsers.FirstOrDefault(du => du.Approved);
+            var approvedDelegateUsers = verifiedDelegateUsers.Where(du => du.Approved).ToList();
 
-            if (verifiedAdminUser == null && approvedDelegateUser == null)
+            if (verifiedAdminUser == null && !approvedDelegateUsers.Any())
             {
                 return View("AccountNotApproved");
             }
 
-            verifiedAdminUser ??= loginService.GetVerifiedAdminUserAssociatedWithDelegateUser(verifiedDelegateUsers.First(), model.Password);
-            
-            LogIn(verifiedAdminUser, approvedDelegateUser, model.RememberMe);
-            return RedirectToAction("Index", "Home");
+            verifiedAdminUser ??=
+                loginService.GetVerifiedAdminUserAssociatedWithDelegateUser(verifiedDelegateUsers.First(), model.Password);
+
+            var availableCentres = GetAvailableCentres(verifiedAdminUser, approvedDelegateUsers);
+
+            if (availableCentres.Count == 1)
+            {
+                return LogIn(verifiedAdminUser, approvedDelegateUsers.FirstOrDefault(), model.RememberMe);
+            }
+
+            TempData.Clear();
+            TempData["RememberMe"] = model.RememberMe;
+            TempData.Set(adminUser);
+            TempData.Set(approvedDelegateUsers);
+
+            return View("ChooseACentre", availableCentres);
+        }
+
+        [HttpPost]
+        public IActionResult ChooseACentre(int centreId)
+        {
+            var rememberMe = (bool)TempData["RememberMe"];
+            var adminAccount = TempData.Get<AdminUser>();
+            var approvedDelegateAccounts = TempData.Get<List<DelegateUser>>();
+
+            var adminAccountForChosenCentre = adminAccount?.CentreId == centreId ? adminAccount : null;
+            var delegateAccountForChosenCentre =
+                approvedDelegateAccounts?.FirstOrDefault(du => du.CentreId == centreId);
+
+            return LogIn(adminAccountForChosenCentre, delegateAccountForChosenCentre, rememberMe);
         }
 
-        private void LogIn(AdminUser? adminUser, DelegateUser? delegateUser, bool rememberMe)
+        private List<ChooseACentreViewModel> GetAvailableCentres(AdminUser? adminUser, List<DelegateUser> delegateUsers)
+        {
+            var availableCentres = new List<ChooseACentreViewModel>();
+
+            foreach (var delegateUser in delegateUsers)

```

Comment:
Could we just map the delegateUser to the available centres here instead to have this more concise (.Select)?